deblur/deblurring.py

```diff
@@ -0,0 +1,167 @@
+        from mean error and sequence length.
+    indel_prob : float, optional
+        Indel probability (same for N indels). Default: 0.01
+    indel_max : int, optional
+        The maximal number of indels expected by errors. Default: 3
+
+    Results
+    -------
+    list of Sequence
+        The deblurred sequences
+
+    Notes
+    -----
+    mean_error is used only for normalizing the peak height before deblurring,
+    whereas read_error is used for calculating the expected number of errors
+    for each position.
+    The array 'error_dist' represents the error distribution, where
+    Xi = max frequency of error hamming. The length of this array - 1 limits
+    the hamming distance taken into account, i.e. if the length if `error_dist`
+    is 10, sequences up to 10 - 1 = 9 hamming distance will be taken into
+    account
+    """
+    # Get the sequences
+    seqs = get_sequences(input_seqs)
+
+    # If mean error is not provided, use the same value as read_error
+    mean_error = mean_error if mean_error else read_error
+
+    # if error_list not supplied, use the default (22 mock mixture setup)
+    mod_factor = pow((1 - mean_error), seqs[0].unaligned_length)
+    if not error_dist:
+        error_dist = np.array(
+            [1.0 / mod_factor, pow(read_error, 1) / mod_factor, 0.01, 0.01,
+             0.01, 0.005, 0.005, 0.005, 0.005, 0.005, 0.005, 0.001, 0.001,
+             0.001, 0.001])
+    else:
+        error_dist = np.array(error_dist) / mod_factor
+
+    max_h_dist = len(error_dist)-1
+
+    for seq_i in seqs:
+        # no need to remove neighbors if freq. is <=0
+        if seq_i.frequency <= 0:
+            continue
+
+        # Correct for the fact that many reads are expected to be mutated
+        num_err = error_dist * seq_i.frequency
+        num_err = [val * seq_i.frequency for val in error_dist]
+
+        # if it's low level, just continue
+        if num_err[1] < 0.1:
+            continue
+
+        # Compare to all other sequences and calculate hamming dist
+        seq_i_len = len(seq_i.sequence.rstrip('-'))
+        for seq_j in seqs:
+            # Ignore current sequence
+            if seq_i == seq_j:
+                continue
+
+            # Calculate the hamming distance
+            h_dist = np.count_nonzero(np.not_equal(seq_i.np_sequence,
+                                                   seq_j.np_sequence))
+
+            # If far away, don't need to correct
+            if h_dist > max_h_dist:
+                continue
+
+            # Close, so lets calculate exact distance
+            num_substitutions = 0
+            num_indels = 0
+
+            # We stop checking in the shortest sequence after removing trailing
+            # indels. We need to do this in order to avoid double counting
+            # the insertions/deletions
+            l = min(seq_i_len, len(seq_j.sequence.rstrip('-')))
+            for bi, bj in zip(seq_i.np_sequence[:l], seq_i.np_sequence[:l]):
```

Comment:
Thanks guys. Just a small correction on muttype:

```
muttype = logical_xor(a[mask] == 4, b[mask] == 4)
```

We want the xor because if they're equal we don't need to count them